### DISCUSSION TITLE
Objective vs. PosteriorTransform

### DISCUSSION THREAD

**Discussion by jduerholt** [meta-pytorch/botorch#2164] [General]
Hi all,

I want to understand in which situations one should use a posterior transform in a single objective acqf and in which an objective module. For me, the functionality looks very much the same, for example when one compares the `ScalarizedPosteriorTransform` to the `LinearMCObjective`. Any advise here?

Best,

Johannes

**Comment by saitcakmak** [comment]
Hi @jduerholt. The `PosteriorTransform`s were originally called something like `AnalyticObjective`, so they're indeed similar. A key difference is that the `PosteriorTransform` operates on the `Posterior` object before sampling from it, while the `MCObjective` operates on the samples drawn from the posterior. This makes it possible to use `PosteriorTransform` with analytic acquisition functions that don't rely on sampling. However, having to analytically transform the `Posterior` object limits their flexibility.

`MCObjective` is quite flexible since it just needs to accept a tensor and return a transformed tensor, so it can support arbitrary callables. The limitation is that before you can use it, you need to sample from the `Posterior`. For seemingly identical transforms like `ScalarizedPosteriorTransform` and `LinearMCObjective`, there's also a slight difference in the computations. `ScalarizedPosteriorTransform` constructs a new posterior object after modifying the mean and mvn, but the sampling from this transformed posterior is cheaper since it has fewer outcomes. `LinearMCObjective` does not require any modifications to the posterior, but it requires sampling from the full posterior and processing those samples after, which can be more expensive & require more memory. I haven't benchmarked it but if I had to pick between these two, I'd lean towards `ScalarizedPosteriorTransform`.

**Reply by jduerholt** [reply]
Hi @saitcakmak, thank you very much for this detailed answer. It made it really clear to me!

**Comment by eytan** [comment]
Curious if folks have any ideas on how we can improve our docs to make this
more clear (and if anybody would like to volunteer to help : )
